pipelines/auth/guarded-repo.yaml

```diff
@@ -0,0 +1,43 @@
+name: Setup auth to clone guarded repos using OctoSTS
+
+needs:
+  packages:
+    - busybox
+    - curl
+    - git
+    - jq
+
+inputs:
+  repo:
+    description: which GitHub repo to clone (e.g., "chainguard-dev/ecosystems-cassandra")
+    required: true
+  identity:
+    description: the identity to use for OctoSTS
+    required: false
+    default: elastic-build
+  cleanup-token:
+    description: whether or not to remove token file after use
+    default: true
+
```

Comment:
I like this but unfortunately, there are a few packages that have build pipelines that end up with multiple `uses: iamguarded/build-compat` steps, so either the `cleanup-token` option needs to be surfaced upwards into that pipeline.